app/assets/javascripts/controllers/cloud_subnet/cloud_subnet_form_controller.js

```diff
@@ -61,27 +53,15 @@ ManageIQ.angular.app.controller('cloudSubnetFormController', ['$http', '$scope',
 
   $scope.filterNetworkManagerChanged = function(id) {
     miqService.sparkleOn();
-
-    $http.get('/cloud_subnet/cloud_subnet_networks_by_ems/' + id)
-      .then(getCloudSubnetNetworksByEmsComplete)
-      .catch(miqService.handleFailure);
-
-    $http.get('/cloud_subnet/cloud_tenants_by_ems/' + id)
-      .then(getCloudTenantsByEmsComplete)
-      .catch(miqService.handleFailure);
-
+    if (id) {
+      API.get("/api/cloud_networks?expand=resources&attributes=name,ems_ref&filter[]=ems_id=" + id).then(function(data) {
```

Comment:
Agreed, the `URL` approach is a bit too verbose, so let's figure out the object thing...

 * `/api` is always there
 * `/collection` is almost always there (`/api/` *is* a valid endpoint we're using)
 * an `/id` is often there
 * potentially followed by another subcollection (and id? ..not sure)

As for query string params..

  * there's usually `expand=resources`
  * lots of places provide a comma-separated list of attributes (`attributes=name,ext_management_system.name,description,cloud_tenant.name,firewall_rules`)
 * some provide a `collection_class` (with the value being an actual ManageIQ::Providers... classname)
 * some provide one or more filters (`filter[]=ems_id=5&filter[]=supports_block_storage?=true`)
 * there's 2 `requester_type=ui` or `ws`, used when logging in
 * and finally there's sorting options `?sort_by=id&sort_order=desc&limit=100`


To me, the biggest pain point with the currently used approach seems to be all those `"?sort_by=" + variable`, becasuse there's no escaping. 

OTOH maybe there's no interesting structure in the URL to simplify, so we could keep that bit as is.

And it should probably know about the little details that multiple attributes means one query param, comma-separated, while multiple filters means multiple params.


So.. what about doing something simple like this:

API.get("/api/collection/{id}", {id: 5}, {
  expand: 'resources',
  sort_by: 'id',
  filter: [ "ems_id=5", "supports_block_storage?=true" ],
  attributes: [ "name", "ext_management_system.name" ],
});

(the assumption being that any {foo} in the url will cause the second param being interpreted as an object of thigns to replace in the url; and the object after that will populate the query string).


As for where we put those API urls, I don't see a reason to pass them around either, we really don't have any place that would be generic enough to talk to multiple different API endpoints in the same way, nor do I think our API is consistent enough to be able to achieve that :).